### PR TITLE
[AIR] Add default prefetch to tf dataset in AIR examples

### PR DESCRIPTION
**Pull request by jiaodong**
Signed-off-by: Jiao Dong [jiaodong@anyscale.com](mailto:jiaodong@anyscale.com)

## Why are these changes needed?

We should recommend better practice in our tf examples.

## Related issue number

#28241

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: doc/source/ray-air/doc_code/tf_starter.py [hunk 1]
```diff
@@ -70,7 +70,9 @@ def to_tensor_iterator():
 
     results = []
     for _ in range(epochs):
-        tf_dataset = to_tf_dataset(dataset=dataset, batch_size=batch_size)
+        tf_dataset = to_tf_dataset(dataset=dataset, batch_size=batch_size).prefetch(
+            tf.data.AUTOTUNE
+        )
```
Comment (amogkam) [inline doc/source/ray-air/doc_code/tf_starter.py:75]: Minor point, but would prefer to have all dataset operations contained inside the `to_tf_dataset` method.

**Comment by amogkam** [comment]
Actually @jiaodong should we just automatically do this in `prepare_dataset_shard`?

**Review by amogkam — APPROVED** [review]

**Comment by amogkam** [comment]
@jiaodong can you ad a sign off for this?

**Comment by jiaodong** [comment]
@amogkam added (i usually just click on the DCO from github ui lol)

**Pull request merged**